awscli/customizations/s3/s3handler.py

```diff
@@ -65,14 +68,16 @@ def __init__(self, session, params, result_queue=None,
         for key in self.params.keys():
             if key in params:
                 self.params[key] = params[key]
-        self.multi_threshold = multi_threshold
-        self.chunksize = chunksize
+        self.multi_threshold = self._runtime_config['multipart_threshold']
+        self.chunksize = self._runtime_config['multipart_chunksize']
+        LOGGER.debug("Using a multipart threshold of %s and a part size of %s",
+                     self.multi_threshold, self.chunksize)
         self.executor = Executor(
-            num_threads=self.EXECUTOR_NUM_THREADS,
+            num_threads=self._runtime_config['max_concurrent_requests'],
```

Comment:
Shouldn't this be
```
num_threads=self._runtime_config['max_concurrent_requests']-1
```

Because the main thread is acting as a producer by calling the ``ListObjects`` operation.